base_layer/p2p/tests/ping_pong.rs

```diff
@@ -0,0 +1,161 @@
+//  Copyright 2019 The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// NOTE: This test uses ports 11111 and 11112
+
+use super::utils::{assert_change, random_string};
+use rand::rngs::OsRng;
+use std::sync::Arc;
+use tari_comms::{
+    connection::NetAddress,
+    connection_manager::PeerConnectionConfig,
+    control_service::ControlServiceConfig,
+    peer_manager::{peer_storage::PeerStorage, NodeIdentity, Peer},
+    types::{CommsDataStore, CommsPublicKey},
+    CommsBuilder,
+};
+use tari_p2p::{
+    ping_pong::{PingPongApiResponse, PingPongService, PingPongServiceApi},
+    services::{ServiceExecutor, ServiceRegistry},
+    tari_message::{NetMessage, TariMessageType},
+};
+use tari_storage::{keyvalue_store::DataStore, lmdb::LMDBBuilder};
+use tempdir::TempDir;
+
+fn new_node_identity(control_service_address: NetAddress) -> NodeIdentity<CommsPublicKey> {
+    NodeIdentity::random(&mut OsRng::new().unwrap(), control_service_address).unwrap()
+}
+
+fn create_peer_storage(tmpdir: &TempDir, name: &str, peers: Vec<Peer<CommsPublicKey>>) -> CommsDataStore {
+    let mut store = LMDBBuilder::new()
+        .set_path(tmpdir.path().to_str().unwrap())
+        .add_database(name)
+        .build()
+        .unwrap();
+
+    store.connect(name).unwrap();
```

Comment:
You should try refactor to the new LMDBStore in the near future

Reply:
#411